tests/functional/splash-example.js

```diff
@@ -0,0 +1,52 @@
+/*globals casper */
+
+'use strict';
+
+casper.test.begin('Test FormatJS home page splash example', function (test) {
+
+    test.comment('Load home page');
+    casper.start(casper.host + '/', function () {
+        test.pass('Page was loaded');
+    });
+
+    casper.then(function () {
+        test.comment('Test splash example');
+
+        test.assertExists('.splash-example-container', 'Found splash example container');
+        test.assertExists('.num-photos-select', 'Found number select');
+        test.assertExists('.locale-select', 'Found locale select');
+        test.assertExists('.splash-example-output', 'Found splash example output');
+
+        var output = casper.evaluate(function (numPhotos, locale) {
+            // Start by setting the takenDate prop to a known value so that we
+            // can compare the example's output to a static value.
+            var container = document.querySelector('.splash-example-container');
+            var component = container.component;
+            component.setProps({ takenDate: Date.now() });
```

Comment:
I could, in theory, not mock the Date object as you suggest, and set the props for each React component I am about to test. However, the prop that matters here is "source", which can be a large amount of code passed as a string, and then eval'ed by the example code. So, in my testData variable, I would have to have that source code. In the end, I think it would make everything more complicated than it needs to be.